### DISCUSSION TITLE
Issue with Localized Fields in child objects for inherited attributes

### DISCUSSION THREAD

**Discussion by AlephLau** [pimcore/pimcore#17706] [Q&A / Support] [answered]
Hi,

I've encountered an issue with Localized Fields in hierarchical objects (parent-child). For example, if a parent product has an attribute like a localized name or description, and it is the same for all child objects, calling getName('en') or getDescription('en') on a child object returns null. However, accessing the same attribute from the parent works as expected.

This requires manual logic to check if the attribute is null in the child and then fetch it from the parent if necessary. If the child has a different value for the localized field, the issue does not occur, and calling getName('en') or similar returns the expected value.

Is this the expected behavior for Localized Fields, or is there a way to make these fields automatically inherit from the parent if they are not set in the child object?

Thanks in advance for your help!

![image](https://github.com/user-attachments/assets/c3b56ac7-a81c-4b06-bef9-cefe75fe2b9a)

**Comment by fashxp** [comment]
No it's not expected, except when you are in Backend-UI context. In what context are you in? Where do you call that code?

**Reply by AlephLau** [reply]
Hi,

Thank you for your response! @fashxp 

To clarify, the request first goes through a controller, which then calls a repository (using Doctrine). In the repository, I retrieve the object, and it's at that point when I try to access the localized field (for example, the name). This is where I encounter the issue—calling getName('en') on the child object returns null, while the parent object works fine.

I'm not working within the Backend-UI context; this is happening in the application logic when fetching data from the repository. I’d greatly appreciate any guidance on how to resolve this behavior or any best practices for handling this type of inheritance for localized fields.

Thanks again for your help!

**Reply by fashxp** [reply]
Just tried my own and it works without problems on my end. 
Make sure you activate the inheritance with `AbstractObject::setGetInheritedValues(true);` (... which should be set automatically in non admin context anyway).
Otherwise, you need to debug and dig a bit to see what the reason is...

**Reply by AlephLau** [reply]
Thanks a lot @fashxp

**Reply by AlephLau** [reply]
When calling my repository from src/Controller/, it works, that is, from App\Controller, but currently, I’m calling it from a bundle I made to extend the functionality of Pimcore admin, and this is where I call my repository and encounter the issue. What can I do to solve this? @fashxp

**Reply by AlephLau** [reply]
I have removed '/admin' from my route '/admin/reviews/list', which was responsible for calling the repository, and now it works.  😄 However, I don’t understand why. I wanted to take advantage of the firewall and role of the '/admin' prefix to design my bundle, but I see it’s only causing me issues. Could you give me any suggestions? Is there something I’m overlooking? @fashxp

**Reply by fashxp** [reply, marked as answer]
When you are in a /admin route, you are in admin context afaik. And then Pimcore sets `AbstractObject::setGetInheritedValues(false);`. 
You need to set it to `true` again...

**Reply by AlephLau** [reply]
Thanks for your time @fashxp  you are awesome!
